onnxoptimizer/passes/eliminate_slice_shape.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+#include "onnx/defs/tensor_util.h"
+#include "onnxoptimizer/pass.h"
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+struct EliminateSliceShape final : public PredicateBasedPass {
+  explicit EliminateSliceShape()
+      : PredicateBasedPass(PassType::Nop, PassEfficiency::Complete,
+                           PassOptimizationType::Compute) {}
+
+  std::string getPassName() const override {
+    return "eliminate_slice_shape";
+  }
+
+  bool patternMatchPredicate(Node *node) override {
+    return node->kind() == kSlice &&
+           node->inputs()[0]->node()->kind() == Symbol("Shape") &&
+           node->inputs()[0]->node()->input()->has_sizes();
+  }
+
+  bool runTransform(Node *node, Graph &graph,
+                    NodeDestroyType &destroy_current) override {
+    Node *shape_node = node->inputs()[0]->node();
+    const auto &dims_of_shape_node_input = shape_node->input()->sizes();
+    std::vector<Dimension> result_of_shape_op;
+
+    auto add_y_if_negative = [](int64_t x, int64_t y) -> int64_t {
```

Comment:
please extract this function into some common file for example creating a new file `onnxoptimizer/passes/utils.h`

Reply:
Other passes also use this helper function, so I will handle with it by new PR.